src/shared/utils/config-utils/fetch-vendor-id.ts

```diff
@@ -0,0 +1,20 @@
+import { fetchVendorServiceConfig } from "./fetch-vendor-service-config";
+
+export const fetchVendorId = async (eventName: string): Promise<string> => {
+  const configBucket = process.env.CONFIG_BUCKET;
+
+  if (configBucket === undefined)
+    throw new Error("No CONFIG_BUCKET defined in this environment");
+
+  // Event name is unique for each vendor id
+  const vendorServiceConfig = (
+    await fetchVendorServiceConfig(configBucket)
```

Comment:
I'm a little worried that we're fetching the vendor service config for each and every event, when we should in theory only need to fetch it once.  I think Peter had a recommendation/example where a similar bit of config was lazy-loaded.  Happy to discuss, maybe at standup?

Reply:
@phinterseer Are you able to share the approach you were talking about in dev stand up?

Reply:
Thanks for raising @markschnitziusgovuk ! I've used Gareth's caching approach from `get-vendor-service-config-row` for now. @Si3rr4wow is writing a tech debt ticket to build a wrapper for all functions that will access the vendor config.